app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.kt

```diff
@@ -0,0 +1,171 @@
+            activity.getString(R.string.no),
+            onPositiveBtnClick,
+            onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String,
+        negativeButtonText: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?
+    ) {
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            positiveButtonText,
+            negativeButtonText,
+            onPositiveBtnClick,
+            onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?,
+        customView: View,
+        cancelable: Boolean
+    ) {
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            activity.getString(R.string.yes),
+            activity.getString(R.string.no),
+            onPositiveBtnClick,
+            onNegativeBtnClick,
+            customView,
+            cancelable
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String?,
+        onPositiveBtnClick: Runnable?,
+        cancellable: Boolean
+    ) {
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            positiveButtonText,
+            onPositiveBtnClick,
+            cancellable
+        )
+    }
+
+    /**
+     * show a dialog
+     * @param activity
+     * @param title
+     * @param message
+     * @param positiveButtonText
+     * @param negativeButtonText
+     * @param onPositiveBtnClick
+     * @param onNegativeBtnClick
+     * @param customView
+     * @param cancelable
+     */
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String? = null,
+        negativeButtonText: String? = null,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?,
+        customView: View? = null,
+        cancelable: Boolean = false
+    ) {
+
+        /* If the custom view already has a parent, there is already a dialog showing with the view
+         * This happens for on resume - return to avoid creating a second dialog - the first one
+         * will still show
+         */
+        if (customView?.parent != null) {
+            return
+        }
+
+        showSafely(activity, AlertDialog.Builder(activity).apply {
+            setTitle(title)
+            setMessage(message)
+            setView(customView)
+            setCancelable(cancelable)
+            positiveButtonText?.let {
+                if (positiveButtonText.isBlank()) {
```

Comment:
you don't have to check `isBlank`, you were only checking that previously because that was your default value.

Lets talk about `let`. 
![image](https://user-images.githubusercontent.com/3358282/78884284-97265480-7a52-11ea-872d-272fc0738a22.png)

[docs](https://kotlinlang.org/docs/reference/scope-functions.html#let)

`inline fun <T, R> T.let(block: (T) -> R): R`

lets break it down, `T.let` this means that anything of Type `T` has a function called `let`, `T` is generic so all objects have a `let` function

`block: (T) -> R` let accepts a parameter called `block`, what this parameter is is a function. that takes a T `(T)` and returns an R `->R`. The easiest way to supply a function in kotlin is just to write `{ }`. We want a function that takes a T so we can write an exampleFunction

```
"myString".let({ myStringAsAVariable -> print(myStringAsAVariable )})
```
but there are some kotlin features to make this less annoying to write.
if the last paramter of a function is another function then we can put it outside the `()`, when the only parameter is a function we can also omit the `()` altogether

```
"myString".let() { myStringAsAVariable -> print(myStringAsAVariable ) }
"myString".let { myStringAsAVariable -> print(myStringAsAVariable ) }
```

when you only have 1 parameter to a lambda ie `myStringAsAVariable` you don't even need to declare it and can instead reference it as `it`

```
"myString".let{ print(it) }
```

Reply:
@macgills Thanks a lot now I understand the use of let.